electron-app/src/services/rotkehlchen-api.ts

```diff
@@ -282,6 +282,22 @@ export class RotkehlchenApi {
     });
   }
 
+  ping(): Promise<boolean> {
+    return new Promise<boolean>((resolve, reject) => {
+      this.axios
+        .get<ActionResult<boolean>>('/ping') // no validate status here since defaults work
+        .then(response => {
+          const { result, message } = response.data;
+          if (result) {
+            resolve(result);
+          } else {
+            reject(new Error(message));
```

Comment:
We don't really care about the errors here, right? In case of an error, we can assume that the ping result is false? If we handle the error case internally in and instead of rejecting we resolve with false you can remove a try-catch in the action

Reply:
Would that catch an internal server error? Would be nice for bug reports right?

Reply:
No, but we were already ignoring errors on the version API before. How many chances there are that the ping API will return a 500? (keep n mind I am not suggesting to change all the APIs only ping)

Reply:
Basically 0 I think -- it just returns true on any get call. On the frontend side I used the approach you took for the version endpoint here, and in that one there is actually a possibility for failure as call to github could possibly fail.

Reply:
An internal server error can always happen.

It can be due to python or something. You never know.

I think it's indeed close to nil chances. But shit happens -- you never know

Reply:
Yes, but how do you want to handle this in the case of connect? Display an error screen if the response is 500? Any other error should be safely ignored. Especially network errors. network errors on the client library just mean that the API is not accessible.

Reply:
so far how do we do this for all other endpoints? I guess the same?

An error window so that the user knows something is really wrong and they can check the logs?

Reply:
> We don't really care about the errors here, right? In case of an error, we can assume that the ping result is false? If we handle the error case internally in and instead of rejecting we resolve with false you can remove a try-catch in the action

If I use the same approach as dsrHistory() as suggested above then this part of the code is replaced by the `handleResponse` method.

Reply:
Yes since everything is doing the same handling no need to repeat that in every method. However, the basic difference is that in the other endpoints we don't ignore network errors. Here we should ignore network errors since we will get a couple of them before the backend is up and running.

Reply:
Apart from an internal server error for some odd reason, the only way `connect` can actually fail is via a network error (i.e. the API is unreachable), which is the one thing we probably should account for. That said, I recall that you had said that the general process for backend calls is that the request is retried for a certain amount of time after which an error is reported, or is that not the case?

Reply:
At the moment the retry is indefinite as you can see in the connect function. The retry stops only on success. We discussed with @LefterisJP before about adding a limit to that but it was never implemented. You can add a limit to retry if you want in this PR. But like I wrote before network errors should be expected, so the ping promise only rejects when there is an API error. On network errors should resolve with false. 

If you add the retry limit you can show the connection error screen either on a retry limit being reached or if the API returns anything but 200.

Reply:
I'm a bit lost; is there an example I can look at? Do we go back from using the handleResponse method and just do it manually?

Which error screen do you mean here, the one where the app totally fails and e.g. shows that the backend has terminated?

Reply:
Perhaps it's better to keep it simple in this PR and just go as is and then make changes if needed?

Reply:
Exactly this screen if ping fails with 500 probably something is really really wrong. I don't have an example sorry, I would also have to figure out how to do that. 

The simplest thing we could do is to leave it with the existing behavior and move any improvements to a new issue.

Reply:
@kelsos In that case should I submit the revised version with `handleResponse` method and infinite retries and then see how to fix it longterm?

Reply:
Yes :) we can create a follow up issue to improve it